File: app/src/main/java/net/squanchy/venue/VenueInfoInjector.java
package net.squanchy.venue;

import android.app.Activity;

import net.squanchy.injection.ActivityContextModule;
import net.squanchy.injection.ApplicationInjector;
import net.squanchy.navigation.NavigationModule;

final class VenueInfoInjector {

    private VenueInfoInjector() {
        // no instances
    }

    public static VenueInfoComponent obtain(Activity activity) {
        return DaggerVenueInfoComponent.builder()
                .applicationComponent(ApplicationInjector.obtain(activity))
                .navigationModule(new NavigationModule())
                .venueInfoModule(new VenueInfoModule())
                .activityContextModule(new ActivityContextModule(activity))
                .build();
    }
}

File: app/src/main/java/net/squanchy/venue/VenueInfoInjector.kt
package net.squanchy.venue

import android.support.v7.app.AppCompatActivity

import net.squanchy.injection.ActivityContextModule
import net.squanchy.injection.ApplicationInjector
import net.squanchy.navigation.NavigationModule

internal object VenueInfoInjector {

    fun obtain(activity: AppCompatActivity): VenueInfoComponent {
        return DaggerVenueInfoComponent.builder()
                .applicationComponent(ApplicationInjector.obtain(activity))
                .navigationModule(NavigationModule())
                .venueInfoModule(VenueInfoModule())
                .activityContextModule(ActivityContextModule(activity))
                .build()
    }
}
